packages/react-core/src/components/Sidebar/__tests__/SidebarPanel.test.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import { SidebarPanel } from '../SidebarPanel';
+
+test('Renders children', () => {
+  render(<SidebarPanel>Test</SidebarPanel>);
+  expect(screen.getByText('Test')).toBeVisible();
+});
+
+test('Renders with class name pf-c-sidebar__panel', () => {
+  render(<SidebarPanel>Test</SidebarPanel>);
+  expect(screen.getByText('Test')).toHaveClass('pf-c-sidebar__panel', { exact: true });
+});
+
+test('Renders with custom class name when className prop is provided', () => {
+  render(<SidebarPanel className="custom-class">Test</SidebarPanel>);
+  expect(screen.getByText('Test')).toHaveClass('custom-class');
+});
+
+test('Renders with class name pf-m-sticky when "sticky" is passed to variant prop', () => {
+  render(<SidebarPanel variant="sticky">Test</SidebarPanel>);
+  expect(screen.getByText('Test')).toHaveClass('pf-m-sticky');
+});
+
+test('Renders with class name pf-m-static when "static" is passed to variant prop', () => {
+  render(<SidebarPanel variant="static">Test</SidebarPanel>);
+  expect(screen.getByText('Test')).toHaveClass('pf-m-static');
+});
+
+test('Renders with class name pf-m-no-background when hasNoBackground prop is passed', () => {
+  render(<SidebarPanel hasNoBackground>Test</SidebarPanel>);
+  expect(screen.getByText('Test')).toHaveClass('pf-m-no-background');
+});
+
+//   test('Renders with default width class names when "default" is passed to each breakpoint of width prop', () => {
+//     render(<SidebarPanel width={{default: "default", sm: "default", md: "default", lg: "default", xl: "default", "2xl": "default", }}>Test</SidebarPanel>);
+//     expect(screen.getByText('Test')).toHaveClass('pf-m-width-default');
+//     expect(screen.getByText('Test')).toHaveClass('pf-m-width-default-on-sm');
+//     expect(screen.getByText('Test')).toHaveClass('pf-m-width-default-on-md');
+//     expect(screen.getByText('Test')).toHaveClass('pf-m-width-default-on-lg');
+//     expect(screen.getByText('Test')).toHaveClass('pf-m-width-default-on-xl');
+//     expect(screen.getByText('Test')).toHaveClass('pf-m-width-default-on-2xl');
+//   });
+
+test('Renders with appropriate width-25 class names when "width_25" is passed to each breakpoint of width prop', () => {
```

Comment:
Rather than having each of these tests written out like this, we could try doing something similar to what the Alert test does by mapping over each possible value for the `width` prop: https://github.com/patternfly/patternfly-react/blob/5d1af989ad4a302d1fa53c4658f033ad0fa1abef/packages/react-core/src/components/Alert/__tests__/Alert.test.tsx#L60-L68

Reply:
I tried to do something similar to what the Alert test was doing, however, I was receiving the following error: 
```
Type '`width_${string}`' is not assignable to type '"default" | "width_25" | "width_33" | "width_50" | "width_66" | "width_75" | "width_100" | undefined'.
```

Reply:
I think the difference is that the `Alert` component has a `AlertVariant` enum while the `SidebarPanel` doesn't have something similar for the different sizes provided to the `width` prop

Reply:
Resolved by adding an enum called `SidebarPanelWidthType`